kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/config/VirtualCluster.java

```diff
@@ -6,19 +6,61 @@
 package io.kroxylicious.proxy.config;
 
 import java.util.List;
+import java.util.Map;
 import java.util.Optional;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 import com.fasterxml.jackson.annotation.JsonProperty;
 
 import io.kroxylicious.proxy.config.tls.Tls;
 
 import edu.umd.cs.findbugs.annotations.Nullable;
 
+/**
+ * A virtual cluster.
+ *
+ * @param targetCluster the cluster being proxied
+ * @param clusterNetworkAddressConfigProvider virtual cluster network config - deprecated - use a named listener
+ * @param tls deprecated - tls settings for the virtual cluster - deprecated - use a named listener
+ * @param listeners listeners
+ * @param logNetwork if true, network will be logged
+ * @param logFrames if true, kafka rpcs will be logged
+ * @param filters filers.
+ */
 public record VirtualCluster(TargetCluster targetCluster,
-                             @JsonProperty(required = true) ClusterNetworkAddressConfigProviderDefinition clusterNetworkAddressConfigProvider,
-                             @JsonProperty() Optional<Tls> tls,
+                             @Deprecated(forRemoval = true, since = "0.11.0") ClusterNetworkAddressConfigProviderDefinition clusterNetworkAddressConfigProvider,
+                             @Deprecated(forRemoval = true, since = "0.11.0") @JsonProperty() Optional<Tls> tls,
+
+                             @JsonProperty(required = false) Map<String, VirtualClusterListener> listeners,
                              boolean logNetwork,
                              boolean logFrames,
                              @Nullable List<String> filters) {
 
+    private static final Logger LOGGER = LoggerFactory.getLogger(VirtualCluster.class);
+
+    @SuppressWarnings("removal")
+    public VirtualCluster {
+        if (clusterNetworkAddressConfigProvider != null || tls.isPresent()) {
+            if (listeners == null || listeners.isEmpty()) {
+                LOGGER.warn(
+                        "The virtualCluster properties 'clusterNetworkAddressConfigProvider' and 'tls' are deprecated, specify virtual cluster listeners using the listeners map.");
```

Comment:
Should include the virtual cluster name in the warning

Reply:
Unfortunately, given we've used map with the virtual cluster name as the key, we can't get it.

Reply:
Raised as #1847